pkgs/development/libraries/zlib/default.nix

```diff
@@ -43,6 +43,10 @@ stdenv.mkDerivation (finalAttrs: {
       hash = "sha256-mpOyt9/ax3zrpaVYpYDnRmfdb+3kWFuR7vtg8Dty3yM=";
     };
 
+  patches = [
+    ./mingw-shared.patch
+  ];
```

Comment:
Would it make sense to make the patch conditional to host platform being mingw? That way this would be zero rebuilds on any other platform i believe